packages/prices-api/src/solver/parsers.ts

```diff
@@ -0,0 +1,27 @@
+import type { Address } from '..'
+import type * as Responses from './responses'
+import type * as Models from './models'
+
+export const parseSolverCompetition = (x: Responses.GetSolverCompetitionResponse): Models.SolverCompetition => ({
+  auctionStartBlock: x.auctionStartBlock,
+  orders: x.auction.orders.map((x) => x),
+  prices: Object.entries(x.auction.prices).reduce<Record<Address, bigint>>((acc, [key, value]) => {
+    acc[key as Address] = BigInt(value)
+    return acc
+  }, {}),
+  solutions: x.solutions.map((sol) => ({
+    solver: sol.solver,
+    solverAddress: sol.solverAddress,
+    score: BigInt(sol.score),
+    ranking: sol.ranking,
+    clearingPrices: Object.entries(sol.clearingPrices).reduce<Record<Address, bigint>>((acc, [key, value]) => {
+      acc[key as Address] = BigInt(value)
+      return acc
+    }, {}),
```

Comment:
We could have a nice helper fn to make these calls clear

```suggestion
    clearingPrices: mapRecord(sol.clearingPrices, (key, value) => [key as Address, BigInt(value)])
```

Reply:
Gets quite hairy wrt typing and generics and I don't want to introduce a lodash dependency. However I've changed the function from using `reduce` to `fromEntries(entries)`, it's a little bit less verbose this way.

Reply:
fromEntries is nice indeed. FYI I created this for another project:
```typescript
export function mapRecord<K extends string, V, R>(obj: Record<K, V>, mapper: (key: K, value: V) => R): Record<K, R> {
  const entries = Object.entries(obj).map(([k, v]) => [k, mapper(k as K, v as V)]);
  return Object.fromEntries(entries) as Record<K, R>;
}
```

Reply:
Will copy this for the next PR like the try/catch that can be removed.